frontend/src/pages/TracesExplorer/Filter/filterUtils.ts

```diff
@@ -0,0 +1,321 @@
+	AllTraceFilterKeyValue,
+) as (keyof typeof AllTraceFilterKeyValue)[];
+
+export const statusFilterOption = ['Error', 'Ok'];
+
+export type FilterType = Record<
+	AllTraceFilterKeys,
+	{ values: string[]; keys: BaseAutocompleteData }
+>;
+
+export const addFilter = (
+	filterType: AllTraceFilterKeys,
+	value: string,
+	setSelectedFilters: Dispatch<
+		SetStateAction<
+			| Record<
+					AllTraceFilterKeys,
+					{ values: string[]; keys: BaseAutocompleteData }
+			  >
+			| undefined
+		>
+	>,
+	keys?: BaseAutocompleteData,
+): void => {
+	setSelectedFilters((prevFilters) => {
+		// If previous filters are undefined, initialize them
+		if (!prevFilters) {
+			return ({
+				[filterType]: { values: [value], keys },
+			} as unknown) as FilterType;
+		}
+		// If the filter type doesn't exist, initialize it
+		if (!prevFilters[filterType]?.values.length) {
+			return {
+				...prevFilters,
+				[filterType]: { values: [value], keys },
+			};
+		}
+		// If the value already exists, don't add it again
+		if (prevFilters[filterType].values.includes(value)) {
+			return prevFilters;
+		}
+		// Otherwise, add the value to the existing array
+		return {
+			...prevFilters,
+			[filterType]: {
+				values: [...prevFilters[filterType].values, value],
+				keys,
+			},
+		};
+	});
+};
+
+// Function to remove a filter
+export const removeFilter = (
+	filterType: AllTraceFilterKeys,
+	value: string,
+	setSelectedFilters: Dispatch<
+		SetStateAction<
+			| Record<
+					AllTraceFilterKeys,
+					{ values: string[]; keys: BaseAutocompleteData }
+			  >
+			| undefined
+		>
+	>,
+	keys?: BaseAutocompleteData,
+): void => {
+	setSelectedFilters((prevFilters) => {
+		if (!prevFilters || !prevFilters[filterType]?.values.length) {
+			return prevFilters;
+		}
+
+		const updatedValues = prevFilters[filterType]?.values.filter(
+			(item) => item !== value,
+		);
+
+		if (updatedValues.length === 0) {
+			const { [filterType]: item, ...remainingFilters } = prevFilters;
+			return Object.keys(remainingFilters).length > 0
+				? (remainingFilters as FilterType)
+				: undefined;
+		}
+
+		return {
+			...prevFilters,
+			[filterType]: { values: updatedValues, keys },
+		};
+	});
+};
+
+export const removeAllFilters = (
+	filterType: AllTraceFilterKeys,
+	setSelectedFilters: Dispatch<
+		SetStateAction<
+			| Record<
+					AllTraceFilterKeys,
+					{ values: string[]; keys: BaseAutocompleteData }
+			  >
+			| undefined
+		>
+	>,
+): void => {
+	setSelectedFilters((prevFilters) => {
+		if (!prevFilters || !prevFilters[filterType]) {
+			return prevFilters;
+		}
+
+		const { [filterType]: item, ...remainingFilters } = prevFilters;
+
+		return Object.keys(remainingFilters).length > 0
+			? (remainingFilters as Record<
+					AllTraceFilterKeys,
+					{ values: string[]; keys: BaseAutocompleteData }
+			  >)
+			: undefined;
+	});
+};
+
+export const traceFilterKeys: Record<
+	AllTraceFilterKeys,
+	BaseAutocompleteData
+> = {
+	durationNano: {
```

Comment:
should these be hardcoded in the UI ?

Reply:
saw some instances already present in code, would cleanup in followup PR